Omit internal types during status logging

pmuens opened the pull request:
This omits our internal types from being logged (such as `AwsProvider`) since those a quite verbose when logging and cause the `JSON.stringify` command to fail due to deeply nested structures.

eahefnawy commented on src/utils/dag/logStatus.js:
```diff
@@ -1,16 +1,25 @@
+        // TODO: replace with a universal util function which checks whether we're
+        // dealing with a Component, Object, Provider, etc. instance here
+        // omitting own Types due to circular structure / verbosity here
+        if (!(value.name && value.version)) {
```
I think it would be more accurate to use the `isType` util method here instead

pmuens replied:
Nice catch and good idea!

pmuens replied:
Unfortunately `isType` doesn't work since it checks for a Type but not for an instance of a Type which is what we're dealing with here.

Additionally smth. like `isComponent` doesn't work since the `Provier` which is passed in via inputs most of the time is not a `Component`, but an `Object`.

eahefnawy commented on src/utils/dag/logStatus.test.js:
```diff
@@ -99,4 +100,36 @@ describe('#logStatus()', () => {
+    logStatus(iteratee, node, context)
+
+    expect(mockLog).toHaveBeenCalledTimes(1)
+    expect(mockLog.mock.calls[0][0]).toMatch(/Deploying.*with parameters.*/)
```
how does this assertion make sure we are not logging complex types? 🤔

pmuens replied:
Yes, that was also something I thought about after pushing. I'll update that :+1

pmuens replied:
Just updated the test so that we have a clear assertion and check to see that we don't output a message about Parameters when we only use the `Provider` as an input.

pmuens commented:
@eahefnawy thanks again for the review. I just updated the PR according to your review.

Additionally I added comments to your review comments for clarification.
